spras/config.py

```diff
@@ -222,16 +223,43 @@ def process_config(self, raw_config):
                 if cur_params[run_params]:
                     for p in cur_params[run_params]:
                         param_name_list.append(p)
-                        all_runs.append(eval(str(cur_params[run_params][p])))
+                        obj = str(cur_params[run_params][p])
+                        try:
+                            obj = [int(obj)]
+                        except ValueError:
+                            try:
+                                obj = [float(obj)]
+                            except ValueError:
+                                # Handles arrays and special evaluation types
+                                # TODO: do we want to explicitly bar `eval` if we may use untrusted user inputs later?
+                                if obj.startswith(("range", "np.linspace", "np.arange", "np.logspace", "[")):
+                                    obj = eval(obj)
+                                elif obj.lower() == "true":
+                                    obj = [True]
+                                elif obj.lower() == "false":
+                                    obj = [False]
+                                else:
+                                    # Catch-all for strings
+                                    obj = [obj]
+                            if not isinstance(obj, Iterable):
+                                raise ValueError(f"The object `{obj}` in algorithm {alg['name']} at key '{p}' in run '{run_params}' is not iterable!") from None
```

Comment:
I was trying to get this error called, but I can't find a combination that would cause this to happen. Do you have a test case where this error is called?

Reply:
This branch is essentially never supposed to get called - any case that a user should reasonably type in should have already been caught by the previous branches. However, it is still possible to get this to be called. For example, try using `range` as a value: since `range` is in the allowlist of objects to be evaluated [for now], `range` is evaluated, but the output of `range` is a function, not an iterable object.

Ideally, we'll have complete parsing support later on that gets rid of `eval` entirely, which we can either handle in this PR or a secondary follow-up one - then, in that world, this case should (ideally) never happen.

Reply:
Ok that makes more sense. Would you be able to add that as a test case in test_config.py?

Reply:
I could, but I don't think it would be much help as a test case - eventually, that behavior won't be desired.